aws/sdk/examples/kinesis/src/bin/put-record.rs

```diff
@@ -3,83 +3,88 @@
  * SPDX-License-Identifier: Apache-2.0.
  */
 
-use std::process;
-
-use kinesis::{Client, Config, Region};
-
-use aws_types::region::{EnvironmentProvider, ProvideRegion};
-
+use aws_types::region::ProvideRegion;
+use kinesis::{Client, Config, Error, Region, PKG_VERSION};
 use structopt::StructOpt;
-use tracing_subscriber::fmt::format::FmtSpan;
-use tracing_subscriber::fmt::SubscriberBuilder;
 
 #[derive(Debug, StructOpt)]
 struct Opt {
-    /// The region
+    /// The default AWS Region.
     #[structopt(short, long)]
-    region: Option<String>,
+    default_region: Option<String>,
 
+    /// The data to add to the stream.
     #[structopt(short, long)]
-    data: String,
+    info: String,
 
+    /// The name of the partition key.
     #[structopt(short, long)]
     key: String,
 
+    /// The name of the stream.
     #[structopt(short, long)]
-    name: String,
+    stream_name: String,
 
     #[structopt(short, long)]
     verbose: bool,
 }
 
+/// Adds a record to an Amazon Kinesis data stream.
+/// # Arguments
+///
+/// * `-s STREAM-NAME` - The name of the stream.
+/// * `-k KEY-NAME` - The name of the partition key.
+/// * `-i INFO` - The data to add.
+/// * `[-d DEFAULT-REGION]` - The Region in which the client is created.
+///    If not supplied, uses the value of the **AWS_REGION** environment variable.
+///    If the environment variable is not set, defaults to **us-west-2**.
+/// * `[-v]` - Whether to display additional information.
 #[tokio::main]
-async fn main() {
+async fn main() -> Result<(), Error> {
+    tracing_subscriber::fmt::init();
+
     let Opt {
-        data,
+        info,
         key,
-        name,
-        region,
+        stream_name,
+        default_region,
         verbose,
     } = Opt::from_args();
 
-    let region = EnvironmentProvider::new()
-        .region()
-        .or_else(|| region.as_ref().map(|region| Region::new(region.clone())))
+    let region = default_region
+        .as_ref()
+        .map(|region| Region::new(region.clone()))
+        .or_else(|| aws_types::region::default_provider().region())
         .unwrap_or_else(|| Region::new("us-west-2"));
 
+    println!();
+
     if verbose {
-        println!("Kinesis client version: {}\n", kinesis::PKG_VERSION);
-        println!("Region:      {:?}", &region);
+        println!("Kinesis version: {}", PKG_VERSION);
+        println!("Region:          {:?}", &default_region);
```

Comment:
I think you want to log the actual region here?

Reply:
Ack, yes. I'll fix it.